modules/gdscript/gdscript_compiler.cpp

```diff
@@ -165,6 +165,173 @@ GDScriptDataType GDScriptCompiler::_gdtype_from_datatype(const GDScriptParser::D
 	return result;
 }
 
+Error GDScriptCompiler::_parse_destructuring(CodeGen &codegen, Error &r_error, const GDScriptParser::DestructuringNode *p_target, GDScriptCodeGenerator::Address p_source) {
+	switch (p_target->type) {
+		case GDScriptParser::Node::ARRAY_DESTRUCTURING: {
+			return _parse_array_destructuring(codegen, r_error, static_cast<const GDScriptParser::ArrayDestructuringNode *>(p_target), p_source);
+		} break;
+		case GDScriptParser::Node::DICT_DESTRUCTURING: {
+			return _parse_dictionary_destructuring(codegen, r_error, static_cast<const GDScriptParser::DictDestructuringNode *>(p_target), p_source);
+		} break;
+		default: {
+			ERR_FAIL_V_MSG(Error::FAILED, "Bug in bytecode compiler, unexpected node in parse tree while parsing expression."); // Unreachable code.
+		} break;
+	}
+}
+
+Error GDScriptCompiler::_parse_array_destructuring(CodeGen &codegen, Error &r_error, const GDScriptParser::ArrayDestructuringNode *p_target, GDScriptCodeGenerator::Address p_source) {
+	int idx = 0;
+	for (GDScriptParser::DestructuringNode *E : p_target->destructured) {
+		if (E->inner_type == GDScriptParser::DestructuringNode::NONE) {
+			// A "none" ("_") destructuring element doesn't need to be computed,
+			// but it does still count towards the next index access.
+			idx += 1;
+			continue;
+		}
+
+		if (E->is_spread) {
+			if (E->inner_type == GDScriptParser::DestructuringNode::DESTRUCTURE) {
+				// Create temp for sliced Array
+				GDScriptCodeGenerator::Address temp = codegen.add_temporary(p_source.type);
+
+				// Slice the Array
+				codegen.generator->write_call(temp, p_source, R"(slice)", { codegen.add_constant(idx) });
+				if (r_error) {
```

Comment:
This hasn't been changed since it was passed as a parameter - it feels like something's missing?

Reply:
Yeah, I probably just copy-pasted that